### ISSUE TITLE
Upgrade  clojure/schema/plumbing dependencies

### ISSUE THREAD

**Issue by mforsyth** [twosigma/satellite#34]
Looks like the AOT issue with Plumbing has been resolved:

https://github.com/Prismatic/plumbing/issues/74 

...so we should be free to upgrade now as long as we're on Clojure 1.7.  

Currently Satellite is at 1.6...
Clojure 1.7 became the stable release this past June 30

**Comment by dgrnbrg** [comment]
We are typically more conservative with upgrading dependencies, only doing so if there's a compelling feature or bugfix.

**Comment by mforsyth** [comment]
@dgrnbrg I don't have a compelling reason at this point.  I had just stumbled across this warning in project.clj
https://github.com/twosigma/satellite/blob/master/satellite-master/project.clj#L45-L48
... and I figured someone must have tried a newer version of schema at some point in the past and gotten burnt.

I'll close this PR.  Can always open a new one if the need arises for some new feature of version 4 or 5 of schema, or something new in Clojure 1.7+.
